core/src/main/java/org/apache/iceberg/ContentFileParser.java

```diff
@@ -27,7 +27,7 @@
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.util.JsonUtil;
 
-class ContentFileParser {
+public class ContentFileParser {
```

Comment:
Why is it necessary to make this constructor public?

Reply:
@rdblue @amogh-jahagirdar 

Currently the `ContentFileParser` is located within package `org.apache.iceberg` as opposed to something like `org.apache.iceberg.rest.responses`. The reason this parser is located within this package is that it is able to access private classes such as `ContentFile`, `GenericDataFile` etc.

However it seems we have placed the `TableScanResponseParser` which is what is used by the majority of endpoints in `org.apache.iceberg.rest.responses`, and this invokes the `ContentFileParser`. When i tried removing the `public` from content file parser naturally got issues.